mithril-aggregator/src/dependency_injection/builder.rs

```diff
@@ -133,8 +133,8 @@ pub struct DependenciesBuilder {
     /// Verification key store.
     pub verification_key_store: Option<Arc<dyn VerificationKeyStorer>>,
 
-    /// Epoch settings storer.
-    pub epoch_settings_storer: Option<Arc<dyn EpochSettingsStorer>>,
+    /// Epoch settings store.
+    pub epoch_settings_store: Option<Arc<EpochSettingsStore>>,
```

Comment:
This is because of the `build_upkeep_service` that use `get_epoch_settings_store` to retrieve the store from the `epoch_settings_store`. 
If it's the trait `EpochSettingsStorer`, it' could not be use as a `EpochPruningTask` even if the trait `EpochSettingsStorer` is a `EpochPruningTask`.
From what I understood, it's a rust limitation with `upcast` and requires adding specific upcast implementations.
https://stackoverflow.com/questions/73003800/how-does-the-trait-upcast-coercion-workaround-work-trait-method-for-getting-an
This seemed acceptable to me because this is only internal variable and functions of the `aggregator` (why `get_epoch_settings_store` is public ?). 
This is also what we do with `SignerStore` and `CertificatePendingStore`.
Do you think we need to rework this part?

Reply:
I'm not sure that the trait casting is a problem :thinking: 

I have created this [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5011d623d14b15f8e42f60430f115a93) that works well by using a `Storer` and casting it to a `Pruner`:
```rust

trait Pruner {
    fn prune(&self){
        println!("prune")
    }
}

trait Storer: Pruner {
    fn store(&self){
        println!("store")
    }
}

struct MyStore;

impl Pruner for MyStore {}
impl Storer for MyStore {}

fn do_something_with_pruner(pruner: &dyn Pruner) {
    println!("do_something_with_pruner");
    pruner.prune();
}

fn main(){
    let store = MyStore{};
    store.store();
    store.prune();
    
    do_something_with_pruner(&store)
}
```

And outputs:
```
store
prune
do_something_with_pruner
prune
```

Reply:
Yes but if your `store` variable is a Arc<dyn Storer> it doesn't work.
You can verify it forcing the type: `let store: Arc<dyn Storer> = Arc::new(MyStore {});`

This is what the `get_epoch_settings_store` returned (`̀Result<Arc<EpochSettingsStore>>`)
I haven't found a simple solution to this problem.

Reply:
It actually works by creating a reference from the `Arc` (with `as_ref()`) as in this [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c0ded17a351fc4e51cce7fc7c32cb58d).
Let me know if that helps :+1: